tools/check_documentation.py

```diff
@@ -0,0 +1,96 @@
+"""Checks for attributes and code examples in RedditBase subclasses"""
+
+import os
+import re
+import sys
+
+# This line imports from the local PRAW rather than the global installed PRAW.
+sys.path.insert(0, os.path.abspath(os.path.join(__file__, "..", "..")))
+
+from praw.models.reddit.base import RedditBase  # noqa: E402
+from praw.models.reddit.modmail import ModmailObject  # noqa: E402
+from praw.util.cache import cachedproperty  # noqa: E402
+
+
+class DocumentationChecker:
+    """Checks for code block statements and attribute tables in subclasses.
+
+    Attribute exceptions holds all exceptions. All classes and subclasses in
+    the exceptions list will be ignored.
+
+    Attribute METHOD_EXCEPTIONS holds the names of methods that will be
+    filtered out.
+    """
+
+    BASE_SEARCH_CLASS = RedditBase
+    exceptions = [
+        ModmailObject,  # is never publicly accessed
+    ]
+    HAS_CODE_BLOCK = re.compile(r".. code-block::")
+    HAS_ATTRIBUTE_TABLE = re.compile(r"Attribute[ ]+Description")
+    METHOD_EXCEPTIONS = [
+        "from_data",
+        "id_from_url",
+        "parse",
+        "sluggify",
+    ]
+    subclasses = []
+
+    @classmethod
+    def expand(cls, base_list, append_list):
+        base_list = list(base_list)
```

Comment:
It appears the intent is to make a copy. Let's not overwrite the variable name in that case.  I'd suggest `to_process` or something along those lines.